### PR TITLE
updated docs role handbook with 1.16 improvements

### PR DESCRIPTION
**Pull request by simplytunde**
I updated the handbook with some of the improvements that were added to the docs release process during 1.16.

@daminisatya  @jimangel @zacharysarah

### CODE REVIEW

File: release-team/role-handbooks/docs/README.md [hunk 1]
```diff
@@ -393,12 +406,11 @@ Create the PRs needed to roll the docs to the new version on the dev branch
 
     24 hours before the release, freeze the repo. No PRs allowed to merge AT ALL until the release PR has successfully merged.
 
-    ⚠️ **NEW:** This hasn't been tested, but after 1.14, we realized that not everyone would see our communications. We can temporarily modify k/website to be a "staging repo" which allows PROW to block all automatic functions (like merging).
+    ⚠️ **NEW:** After 1.16 release, we realized you only need to create an issue with ```blocker_label``` to prevent merging on a repo. Please check with **test-infra**  as this is not tested. You can find more information [here](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/tide/config.md#merge-blocker-issues)
```
Comment (sftim) [inline release-team/role-handbooks/docs/README.md]: What does ```blocker_label``` mean here?

Comment (simplytunde) [reply release-team/role-handbooks/docs/README.md]: It means the label. I will update to make it clearer.

**Comment by tengqm** [comment]
@simplytunde We may want to one extra step for fixing the deprecation warning issue as we discussed [here](https://github.com/kubernetes/website/pull/16131).

**Comment by tengqm** [comment]
Thank you. 
/lgtm

**Pull request merged**